examples/composition/methods/guest/src/main.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use hello_world_methods::MULTIPLY_ID;
+use risc0_zkvm::{guest::env, serde};
+
+fn main() {
+    let n: u64 = env::read();
+    let e: u64 = env::read();
+
+    // x value that will be kept private.
+    let x: u64 = env::read();
+
+    // Verify that n has a known factorization.
+    env::verify(
+        MULTIPLY_ID,
+        bytemuck::cast_slice(&serde::to_vec(&n).unwrap()),
```

Comment:
I feel like users requiring to use `bytemuck` is a code-smell.

Reply:
Resolved this by accepting `&[impl Pod]` in the `env::verify` signature.